### DISCUSSION TITLE
Paragraphs in a Grid Table PDF output seems to lack vertical spacing (compared to HTML)

### DISCUSSION THREAD

**Discussion by fuhrmanator** [quarto-dev/quarto-cli#3621] [Q&A] [answered]
Can anyone say why paragraphs rendered inside of Grid Tables don't have space after them (as they do outside of Grid Tables, or when rendered to HTML). Here's a MWE:

PDF:
![image](https://user-images.githubusercontent.com/7606540/206797729-ffdaec20-fe9e-4643-890b-0bd4848c8efc.png)

HTML:
![image](https://user-images.githubusercontent.com/7606540/206797791-4381a09d-6ab6-499e-b5c0-01d223ac35ea.png)

Markdown source:
```markdown
Grid table problem with cells containing paragraphs (HTML vs PDF)

+----------------------------------------+
| Col1                                   |
+========================================+
| Here's one paragraph                   |
|                                        |
| Here's a second one in the same cell.  |
+----------------------------------------+
| Here's a paragraph followed by a list: |
|                                        |
| -   one                                |
|                                        |
| -   two                                |
|                                        |
| -   three                              |
+----------------------------------------+

: Here's a sample table caption
```

I'm using RStudio's Visual editor for Quarto to make the table.

I can correct the PDF by adding hard line breaks (Ctrl-Enter), but then the HTML version has extra lines. 

I'm wondering if this is a problem with Pandoc and Grid Tables (I quickly searched with Google but didn't find anything).

**Comment by baptiste** [comment, marked as answer]
It's a Latex quirk, so you can try some solutions found in Latex forums (searching for \parksip inside tabular). Here's [one possible workaround](https://tex.stackexchange.com/questions/279207/why-is-parskip-zero-inside-a-tabular),

```
---
title: "Untitled"
format: 
  pdf:
    keep-tex: true
    include-in-header: 
      text: |
         \usepackage{array}
         \setlength{\parindent}{0mm}
         \setlength{\parskip}{8pt}
         \makeatletter
         \def\@arrayparboxrestore{%
         \let\if@nobreak\iffalse
         \let\if@noskipsec\iffalse
         \let\par\@@par
         \let\-\@dischyph
         \let\'\@acci\let\`\@accii\let\=\@acciii
         \everypar{}%
         \linewidth\hsize
         \@totalleftmargin\z@
         \leftskip\z@skip \rightskip\z@skip \@rightskip\z@skip
         \parfillskip\@flushglue \lineskip\normallineskip
         \baselineskip\normalbaselineskip
         \sloppy}
         \makeatother
---
```

(you'd probably want to hide it in a separate file)

**Reply by fuhrmanator** [reply]
I put the [second answer](https://tex.stackexchange.com/a/279280/17868) inside the [`template.tex` (which is the full latex template)](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/pdf/pandoc/template.tex) and pointed to it in the YAML using [`format:/pdf:/template:`](https://quarto.org/docs/journals/templates.html#templates). It solved the problem quite nicely. 

Another inline solution that is compatible with HTML is to put `\newline` which is passed to LaTeX only by pandoc under the hood (or so I understand). This is actually nicer because the `parskip` fix in the template only works for spacing between subsequent paragraphs. There is no space after the last paragraph in a cell, which, again, looks strange to me compared to the HTML layout.
